connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -257,51 +256,47 @@ ClusterAssignment performTaskAssignment(
-        log.debug("Complete (ignoring deletions) connector assignments: {}", connectorAssignments);
-
-        // Per worker task assignments without removing deleted connectors yet
-        Map<String, Collection<ConnectorTaskId>> taskAssignments =
-                completeWorkerAssignment.stream().collect(Collectors.toMap(WorkerLoad::worker, WorkerLoad::tasks));
-        log.debug("Complete (ignoring deletions) task assignments: {}", taskAssignments);
+        ConnectorsAndTasks created = diff(configured, previousAssignment, activeAssignments);
+        log.debug("New assignments: {}", created);
 
         // A collection of the current assignment excluding the connectors-and-tasks to be deleted
         List<WorkerLoad> currentWorkerAssignment = workerAssignment(memberAssignments, deleted);
 
-        Map<String, ConnectorsAndTasks> toRevoke = computeDeleted(deleted, connectorAssignments, taskAssignments);
-        log.debug("Connector and task to delete assignments: {}", toRevoke);
+        Map<String, ConnectorsAndTasks.Builder> toRevoke = new HashMap<>();
 
-        // Revoking redundant connectors/tasks if the workers have duplicate assignments
-        toRevoke.putAll(computeDuplicatedAssignments(memberAssignments, connectorAssignments, taskAssignments));
-        log.debug("Connector and task to revoke assignments (include duplicated assignments): {}", toRevoke);
-
-        // Recompute the complete assignment excluding the deleted connectors-and-tasks
-        completeWorkerAssignment = workerAssignment(memberAssignments, deleted);
-        connectorAssignments =
-                completeWorkerAssignment.stream().collect(Collectors.toMap(WorkerLoad::worker, WorkerLoad::connectors));
-        taskAssignments =
-                completeWorkerAssignment.stream().collect(Collectors.toMap(WorkerLoad::worker, WorkerLoad::tasks));
+        Map<String, ConnectorsAndTasks> deletedAndRevoked = intersection(deleted, memberAssignments);
+        log.debug("Deleted connectors and tasks to revoke from each worker: {}", deletedAndRevoked);
+        addAll(toRevoke, deletedAndRevoked);
 
-        handleLostAssignments(lostAssignments, newSubmissions, completeWorkerAssignment);
+        // Revoking redundant connectors/tasks if the workers have duplicate assignments
+        Map<String, ConnectorsAndTasks> duplicatedAndRevoked = intersection(duplicated, memberAssignments);
+        log.debug("Duplicated connectors and tasks to revoke from each worker: {}", duplicatedAndRevoked);
+        addAll(toRevoke, duplicatedAndRevoked);
+
+        // Compute the assignment that will be applied across the cluster after this round of rebalance
+        // Later on, new submissions and lost-and-reassigned connectors and tasks will be added to these assignments,
+        // and load-balancing revocations will be removed from them.
+        List<WorkerLoad> nextWorkerAssignment = workerLoads(memberAssignments);
```

Comment:
`nextWorkerAssignment` basically signifies the workerAssignments for this round, right? It has `next` because it eventually gets assigned to `previous` after the rebalance round?

Reply:
That's correct, `nextWorkerAssignment` is the assignment that will be given out after this round completes. There's three kinds of assignment we care about during this round:
- The assignment that we sent out during the last round (`previousAssignment`)
- The assignment that everyone has reported in their metadata while joining the group during the current rebalance (`currentWorkerAssignment`)
- The assignment that we'll give out at the end of the current rebalance (`nextConnectorAssignments`)